test/e2e/e2e_test.go

```diff
@@ -292,20 +295,28 @@ func createPod(f *framework.Framework, podName, nodeSelector, namespace string,
 			RestartPolicy: v1.RestartPolicyNever,
 		},
 	}
+
+	for _, o := range options {
+		o(pod)
+	}
+
 	podClient := f.ClientSet.CoreV1().Pods(namespace)
-	_, err := podClient.Create(pod)
+	res, err := podClient.Create(pod)
 	if err != nil {
 		framework.Logf("Warning: Failed to get logs from pod %q: %v", pod.Name, err)
 	}
-	err = e2epod.WaitForPodNotPending(f.ClientSet, podName, namespace)
-	if err != nil {
-		logs, logErr := e2epod.GetPodLogs(f.ClientSet, namespace, pod.Name, contName)
-		if logErr != nil {
-			framework.Logf("Warning: Failed to get logs from pod %q: %v", pod.Name, logErr)
-		} else {
-			framework.Logf("pod %s/%s logs:\n%s", namespace, pod.Name, logs)
-		}
-	}
+	err = e2epod.WaitForPodNotPending(f.ClientSet, namespace, podName)
+	framework.ExpectNoError(err)
```

Comment:
Will this short-circut (and thus prevent log gathering?)

Reply:
I think so, but will the pod log if it never gets running?
My understanding is that the code below is to dump logs for healthy pods. That was the same behaviour that was in place previously, I just re-arranged it and added the active waiting for logs to appear (or, for the get logs not to fail).

Reply:
@squeed now I also retroactively understand this comment